demo_callbacks.py

```diff
@@ -0,0 +1,347 @@
+        shifts_per_employees: The min and max shifts each part-time employee should be scheduled for.
+        scenario: The scenario preset that is selected.
+        custom_saved_data: The saved custom scenario data to update.
+
+    Returns:
+        custom-saved-data: The saved custom scenario data to update.
+    """
+    if scenario == 0:
+        custom_saved_data.update({ctx.triggered_id: ctx.triggered[0]["value"]})
+        return custom_saved_data
+
+    raise PreventUpdate
+
+
+@dash.callback(
+    Output("availability-content", "children"),
+    Output("schedule-content", "children"),
+    Output("schedule-tab", "disabled"),
+    Output("tabs", "value"),
+    Output({"type": "to-collapse-class", "index": 1}, "style"),
+    Output({"type": "forecast", "index": ALL}, "value"),
+    Output({"type": "forecast", "index": ALL}, "placeholder"),
+    Output({"type": "forecast", "index": ALL}, "max"),
+    Output("num-full-time-select", "max"),
+    Output("num-full-time-select", "marks"),
+    Output("num-full-time-select", "value"),
+    inputs=[
+        Input("num-employees-select", "value"),
+        Input("num-full-time-select", "value"),
+    ],
+)
+def display_initial_schedule(
+    num_employees: int, num_full_time: int
+) -> tuple[pd.DataFrame, pd.DataFrame, bool, str, dict, list[dict]]:
+    """Display initial availability schedule.
+
+    Display initial schedule in, and switch to, the availability
+    tab if number of employees has changed.
+
+    Args:
+        num_employees: The number of employees.
+        num_full_time: The number of full-time employees.
+
+    Returns:
+        availability-content: The availability tab content.
+        schedule-content: The schedule tab content.
+        schedule-tab-disabled: Whether the schedule tab should be disabled.
+        tabs-value: The tab that should be selected.
+        to-collapse-class-style: The style for the errors tab.
+        forecast: The forecasted employees per shift requirements value.
+        forecast: The forecasted employees per shift requirements placeholder.
+        forecast: The forecasted employees per shift requirements max.
+        num-full-time-select-max: The max to set the full-time select to.
+        num-full-time-select-marks: The marks to set for the full-time select.
+        num-full-time-select-value: The value to set for the full-time select.
+    """
+    new_full_time_max = math.floor(num_employees * 3 / 4)
+    full_time_marks = {
+        NUM_FULL_TIME["min"]: str(NUM_FULL_TIME["min"]),
+        new_full_time_max: str(new_full_time_max),
+    }
+    num_full_time = min(num_full_time, new_full_time_max)
+
+    df = utils.build_random_sched(num_employees, num_full_time)
+
+    init_availability_table = utils.display_availability(df)
+
+    # Prepare forecast defaults
+    df_to_count = df.iloc[:num_full_time, :]
+    count = df_to_count.applymap(lambda cell: cell.count(REQUESTED_SHIFT_ICON)).sum()[1:].to_dict()
```

Comment:
`df.applymap` has been deprecated in favour of `df.map`.
```suggestion
    count = df_to_count.map(lambda cell: cell.count(REQUESTED_SHIFT_ICON)).sum()[1:].to_dict()
```